### ISSUE TITLE
ScrollView Zoom Bug

### ISSUE THREAD

**Issue by mitran79** [microsoft/microsoft-ui-xaml#1469]
**Describe the bug**
- Unhandled win 32 exception thrown.
- ScrollView Excessive memory consumption. From 250MB up to 10GB upon zooming about.

**Steps to reproduce the bug**

1. Run app.
2. Zoom in and out real quick with mouse on a shape. Tends to happen when zooming in on shape.

**Expected behavior**

- Normal memory consumption.
- No exception thrown.

**Screenshots**

![Screenshot (17)](https://user-images.githubusercontent.com/36862623/67169064-bc67cd80-f3f4-11e9-9988-3db1acc6ca10.png)
![Screenshot (18)](https://user-images.githubusercontent.com/36862623/67168716-93ded400-f3f2-11e9-9f43-8a778db74fca.png)
![Screenshot (19)](https://user-images.githubusercontent.com/36862623/67168718-980af180-f3f2-11e9-83ef-41e76ea908e7.png)
![Screenshot (20)](https://user-images.githubusercontent.com/36862623/67168722-9b05e200-f3f2-11e9-80cf-dc645b1c5019.png)
**Version Info**

Windows 10, version 1903 (10.0; Build 18362)
NuGet package version: 

Microsoft.NETCore.UniversalWindowsPlatform 6.2.9
WindowsRuntime 1.4;CLR v4.0.30319
UAP,Version=10.0.18362.0

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2019 Update (18362)            | YES   |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | YES |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

Please see attached VS2017-2019 solution zip.
[ScrollViewZoomBug.zip](https://github.com/microsoft/microsoft-ui-xaml/files/3748592/ScrollViewZoomBug.zip)

**Comment by robloo** [comment]
I ran across what I think is a similar issue a few years ago with the Canvas control instead of the ScrollViewer. I created a user voice bug report that was never addressed -- I can't find that anymore as I think they are no longer accessible. If my notes are correct this appeared in the Fall Creators Update. Before that update things were rendered very pixelated at these zoom levels.

I wrote a zoom-able graph using a Canvas, PathFigure, Lines, TextBlocks, ViewBoxes, etc. as below. Zooming in past 300% can crash just as you described. To work around this I have to artificially limit the zoom percentage.

![image](https://user-images.githubusercontent.com/17993847/67201250-e3fa6e00-f3d3-11e9-85e3-71f239f3ddba.png)

![image](https://user-images.githubusercontent.com/17993847/67201276-f1aff380-f3d3-11e9-9cec-3b36078a39fe.png)

crash...

"An unhandled win32 exception occured in [8300] ..." The exception is native.
